src/realm/util/fifo_helper.cpp

```diff
@@ -0,0 +1,95 @@
+/*************************************************************************
+ *
+ * Copyright 2019 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **************************************************************************/
+
+#include <realm/util/fifo_helper.hpp>
+
+#include <sstream>
+#include <system_error>
+#include <sys/stat.h>
+
+namespace realm {
+namespace util {
+
+void create_fifo(std::string path, const std::string tmp_dir)
+{
+#ifdef _WIN32
+    throw std::logic_error("mkfifo is not supported on Windows");
+#else
+#ifdef REALM_ANDROID
+    // Upgrading apps on Android Huawai devices sometimes leave FIFO files with the wrong
+    // file owners. This results in the Android sandbox preventing Realm from opening the
+    // database file. To prevent this from happening we create all FIFO files with full permissions.
+    // This should be safe as the app is already running inside a protected part of the filesystem,
+    // so no outside users have access to the filesystem where these files are.
+    // It does make storing Realms on external storage slightly more unsafe, but users already have
+    // full access to modify or copy files from there, so there should be no change from a pratical
+    // security standpoint.
+    // See more here: https://github.com/realm/realm-java/issues/3972#issuecomment-313675948
+    mode_t mode = 0600;
```

Comment:
Out of curiousity, are these two values meant to be the same?

Reply:
No, it is supposed to be 0x666 (I think), right now I'm just trying to make it compile :)

Reply:
Might be worth trying 888, I remember that being the "accessible by everyone" and as this is internal storage, it should be ok